pkg/streamingpromql/testutils/utils.go

```diff
@@ -20,6 +20,13 @@ import (
 // It's possible that floating point values are slightly different due to imprecision, but require.Equal doesn't allow us to set an allowable difference.
 func RequireEqualResults(t testing.TB, expr string, expected, actual *promql.Result, skipAnnotationComparison bool) {
 	require.Equal(t, expected.Err, actual.Err)
+
+	if expected.Err != nil && actual.Err != nil && expected.Err.Error() == actual.Err.Error() {
```

Comment:
This should be sufficient:

```suggestion
	if expected.Err != nil {
```

We've already checked that `expected.Err` and `actual.Err` are equal above, so we only need to check if there's an error.